### PR TITLE
Add section for distributing pipeline best practice

### LINKED ISSUE #819: Add basic best practice documentation for distributing pipelines
As #608 points out, distributing/sharing pipelines can be problematic due to the current implementation.  We should produce minimal documentation that helps users navigate this scenario, but will address this more fully in a future release.

### PR DESCRIPTION
**Pull request by kevin-bates**
This adds a couple of paragraphs describing best practices for sharing/distributing pipelines after first describing the component relationships.

I also moved the developer-specific pipeline sections (file format, etc.) to a new Pipelines topic in the Developer Guide and updated its fields to correspond to version 1, as well as fixed a non-fatal build issue.

Resolves #819 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -88,7 +88,7 @@
 # further.  For a list of options available for each theme, see the
 # documentation.
 html_theme_options = {
-    'description': "Elyra extends JupyterLab Notebooks with an AI centric approach",
+#    'description': "Elyra extends JupyterLab Notebooks with an AI centric approach",
```
Comment (lresende) [inline docs/source/conf.py:91]: Curious to what is the issue here?

File: docs/source/user_guide/pipelines.md [hunk 2]
```diff
@@ -66,167 +66,32 @@ needed for the pipeline are present.
...
+
+#### Distributing Pipelines - Best Practice
+Prior to distributing your pipeline - which includes preserving the component relationships - it is best to commit these files (and directories) to a github repository.  An alternative approach would be to archive the files using `tar` or `zip`, but, again preserving the component relationships.
+
+When deploying a shared or distributed pipeline repository or archive, it is **very important** that the pipeline notebooks be extracted into the **same directory hierarchy relative to the hosting notebook server's working directory**.
+
+
+For example, if you archived your pipeline into a github repository named `my_analytics_pipeline` from the notebook server's working directory, then consumers of this archive would need to do the following:
+
+1. Clone your repository
```
Comment (lresende) [inline docs/source/user_guide/pipelines.md:83]: I believe there are two options/best practices here: 
1 - The one you describe, which requires one single project per workspace
2 - publishing your project inside a folder, probably the same name as the repo, which should allow us to have multiple github projects in one workspace

Thoughts?  Should we also mention the number 2 ?

File: docs/source/conf.py (hunk 1 above)
Comment (kevin-bates) [reply docs/source/conf.py:91]: 'description' isn't a valid html_theme option and the following is produced:
```
preparing documents... WARNING: unsupported theme option 'description' given
```

File: docs/source/user_guide/pipelines.md (hunk 2 above)
Comment (kevin-bates) [reply docs/source/user_guide/pipelines.md:83]: Yes - I agree.  It's all a matter of capturing the project relative to the notebook directory.  Thanks for the suggestion.

File: docs/source/conf.py (hunk 1 above)
Comment (kevin-bates) [reply docs/source/conf.py:91]: @lresende - I left it commented out in case there's another place it should go, but would prefer it be removed.

Comment (lresende) [reply docs/source/conf.py:91]: We can remove it.

**Pull request merged**